quadro_solve.cpp

```diff
@@ -0,0 +1,97 @@
+/**
+ * @brief This file contains functions that calculate the number of roots of a quadratic equation
+ * and their values, if any exist
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include <assert.h>
+double DBL_EPSILON = 1e-6;
+
+enum NumOfSolutions {
+    INF_SOLUTIONS = -1,
+    ZERO_SOLUTIONS = 0,
+    ONE_SOLUTION = 1,
+    TWO_SOLUTIONS = 2
+};
+
+enum TrueFalse {
+    TRUE = 1,
+    FALSE = 0
+};
+
+/**
+ * @brief A function that compares double with zero
+ * @param term
+ * @return TrueFalse
+ */
+
+unsigned short zeroComprasion(double term) {
+    if (fabs(term) < DBL_EPSILON) {
+        return TRUE;
+    }
+    return FALSE;
+}
+
+/**
+ * @brief A function that counts the value of x for two specified variables
+ * @param x_coef
+ * @param free_term
+ * @param x the result will be here
+ * @return NumOfSolutions
+ */
+
+int solveLinear(double x_coef, double free_term, double* x) {
+    assert(x != NULL);
+    assert(zeroComprasion(x_coef));
+    *x = -free_term / x_coef;
+    return ONE_SOLUTION;
+}
+
+/** \brief A function that returns the number of roots of a square equation and their values
+    \version 1.0.0
```

Comment:
If you want to include \version to your documentation, then add it for all functions.